go/cmd/dolt/commands/remote.go

```diff
@@ -230,3 +242,81 @@ func printRemotes(dEnv *env.DoltEnv, apr *argparser.ArgParseResults) errhand.Ver
 
 	return nil
 }
+
+func pingRemote(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgParseResults) errhand.VerboseError {
+	if apr.NArg() != 2 {
+		return errhand.BuildDError("").SetPrintUsage().Build()
+	}
+
+	name := strings.TrimSpace(apr.Arg(1))
+	remote, ok := dEnv.RepoState.Remotes[name]
+	if !ok {
+		return errhand.BuildDError("error: unknown remote: '%s' ", name).Build()
+	}
+
+	meta, err := getRemoteMetadata(ctx, dEnv, remote)
+	if err != nil {
+		return errhand.VerboseErrorFromError(err)
+	} else if meta == nil {
+		cli.PrintErrf("failed to get metadata for %s\n", remote.Name)
+	}
+
+	cli.Printf("%s (%s)\n", remote.Name, remote.Url)
+	cli.Printf("\tNbfVersion:  '%s'\n", meta.NbfVersion)
+	cli.Printf("\tNbsVersion:  '%s'\n", meta.NbsVersion)
+	cli.Printf("\tStorageSize: %d\n", meta.StorageSize)
+	return nil
+}
+
+func getRemoteMetadata(ctx context.Context, dEnv *env.DoltEnv, remote env.Remote) (*remotesapi.GetRepoMetadataResponse, error) {
+	url, err := earl.Parse(remote.Url)
```

Comment:
Kinda confused here...

Why isn't this something like,

```go
ddb, err := env.GetRemoteDB(ctx, nbf, env)
if err != nil {
   ...
}
db := doltdb.HackDatasDatabaseFromDoltDB(ddb)
remotenbf := db.Format()
tfformat := datas.ChunkStoreFromDatabase(db).Version()
```

And, if you need check if remote exists, something like `datas..ChunkStoreFromDatabase(db).Root(ctx) != hash.Hash{}`?